prov/tcp/src/tcpx_init.c

```diff
@@ -44,20 +44,44 @@
 #if HAVE_GETIFADDRS
 static void tcpx_getinfo_ifs(struct fi_info **info)
 {
+	char *tcpx_interface_name = NULL;
 	struct ifaddrs *ifaddrs, *ifa;
 	struct fi_info *head, *tail, *cur, *loopback;
 	size_t addrlen;
 	uint32_t addr_format;
 	int ret;
 
+	fi_param_get_str(&tcpx_prov, "iface", &tcpx_interface_name);
+
 	ret = ofi_getifaddrs(&ifaddrs);
 	if (ret)
 		return;
 
 	head = tail = loopback = NULL;
+	if (tcpx_interface_name) {
+		for (ifa = ifaddrs; ifa != NULL; ifa = ifa->ifa_next) {
+			if (strncmp(tcpx_interface_name, ifa->ifa_name,
+				    strlen(tcpx_interface_name)) == 0) {
+				break;
+			}
+		}
+		if (ifa == NULL) {
+			FI_INFO(&tcpx_prov, FI_LOG_CORE,
+				"Can't set filter to unknown interface: (%s)\n",
+				tcpx_interface_name);
+			tcpx_interface_name = NULL;
+		}
+	}
```

Comment:
If the user specifies the wrong name, we shouldn't accept any name as valid.  It would be better to accept none, so that the failure is visible, and the user can correct the error.  I would drop the above code and just use the checks below.

Reply:
The above code were removed